### ISSUE TITLE
Better displayNames for React DevTools

### ISSUE THREAD

**Issue by SMLMRKHLMS** [emotion-js/emotion#358]
Love this library and I am currently working on a POC to persuade my team and organization to use it in production. Thanks!

- `emotion` version: 7.3.2
- `react` version: 15.6.1

Problem description:
Currently, React DevTools shows emotion components like so: `styled(Component)`. Is there an option in the babel plugin to transform the displayName into something more debuggable/readable. 

Suggested solution:
I would prefer if the displayName was transformed to whatever the name of the component is when the variable is declared. Cheers!

**Comment by greggb** [comment]
For my Storybook library I've been reassigning so the source is more straightforward for non-devs. This may help while putting together your POC for now. In this example the Storybook info addon will display `AnyChildElement` for these columns rather than `styled`.

```
const DemoColumn = styled.div`
    background: ${colorMap.slateLight};
    padding: ${spacing[1]};
`;

const AnyChildElement = () => <DemoColumn />;
```

**Comment by avaragado** [comment]
@mitchellhamilton I realise this issue is closed with a fix, but I suspect it's not the solution @SMLMRKHLMS was looking for. Consider this component:

```js
const Circle = styled.div`
    width: 100px;
    height: 100px;
    border-radius: 50%;
    background-color: red;
`;

export default Circle;
```

In React DevTools, this now shows up as `Styled(div)` (in react-emotion 8.0.2-13). Without digging into code you can't see that it's a `Circle`. This makes debugging etc trickier.

A workaround: set the `displayName` property of the exported function.

```js
Circle.displayName = 'Circle';
```

**Comment by mitchellhamilton** [comment]
Yup, @tkh44 is working on this in #375.

**Comment by mitchellhamilton** [comment]
Closed via #375 and #442 

[Use the `autoLabel` option in your config for `babel-plugin-emotion`](https://github.com/emotion-js/emotion/tree/master/packages/babel-plugin-emotion#autolabel) and the identifier name will be on the displayName.
